setuptools/build_meta.py

```diff
@@ -26,53 +26,38 @@
 Again, this is not a formal definition! Just a "taste" of the module.
 """
 
-import io
 import os
 import sys
 import tokenize
 import shutil
 import contextlib
 import tempfile
 import warnings
+from itertools import chain
+from uuid import uuid4
 
 import setuptools
 import distutils
 from ._reqs import parse_strings
+from setuptools.extern.packaging.requirements import Requirement
 
 __all__ = ['get_requires_for_build_sdist',
            'get_requires_for_build_wheel',
            'prepare_metadata_for_build_wheel',
            'build_wheel',
            'build_sdist',
-           '__legacy__',
-           'SetupRequirementsError']
+           '__legacy__']
 
-
-class SetupRequirementsError(BaseException):
-    def __init__(self, specifiers):
-        self.specifiers = specifiers
-
-
-class Distribution(setuptools.dist.Distribution):
-    def fetch_build_eggs(self, specifiers):
-        specifier_list = list(parse_strings(specifiers))
-
-        raise SetupRequirementsError(specifier_list)
-
-    @classmethod
-    @contextlib.contextmanager
-    def patch(cls):
-        """
-        Replace
-        distutils.dist.Distribution with this class
-        for the duration of this context.
-        """
-        orig = distutils.core.Distribution
-        distutils.core.Distribution = cls
-        try:
-            yield
-        finally:
-            distutils.core.Distribution = orig
+SETUP_SCRIPT = "setup.py"
+CONFIG_FILE = "pyproject.toml"
+LEGACY_CONFIG_FILE = "setup.cfg"
+SAFE_SCRIPT_NAME = "%%setuptools.build_meta%%"
```

Comment:
If this needs to be non-existing, why not add a random factor to it, like `uuid.uuid4()`?